code/game/objects/items/stunbaton.dm

```diff
@@ -110,7 +125,20 @@
 			cell = W
 			to_chat(user, span_notice("You install a cell in [src]."))
 			update_icon()
-
+	else if(istype(W, /obj/item/firing_pin))
+		if(pin)
+			to_chat(user, span_notice("[src] already has a firing pin. You can remove it with crowbar"))
+	else if(W.tool_behaviour == TOOL_CROWBAR)
+		if(pin)
+			pin.forceMove(get_turf(src))
+			pin = null
+			status = FALSE
+			to_chat(user, span_notice("You remove the firing pin from [src]."))
+		if(upgrade)
```

Comment:
remove this, make the upgrade just change the stats once, and then delete the upgrade.
nobody is going to un-upgrade their baton

Reply:
They might want to due to the extra power draw.

Reply:
thats the tradeoff

Reply:
As batons are not easily obtainable I cant make them disposable like that.